match/lib/match/spaceship_ensure.rb

```diff
@@ -50,6 +50,34 @@ def certificate_exists(username: nil, certificate_id: nil)
       UI.user_error!("To reset the certificates of your Apple account, you can use the `match nuke` feature, more information on https://github.com/fastlane/fastlane/tree/master/match")
     end
 
+    def certificate_exists_for_pkcs12(pkcs12)
+      UI.message "Looking up availabe code signing certificates in the Apple Developer Portal..."
+
+      matching_certificates_on_portal = Spaceship.certificate.all.find_all do |cert|
+        (cert.expires == pkcs12.certificate.not_after)
+      end
+
+      if !matching_certificates_on_portal.nil? and matching_certificates_on_portal.count > 1
+        UI.warning "Found more than one eligible certificate in the Developer Portal that matches the certificate to import, trying to match on `owner_id` too..."
+
+        cert_cn = pkcs12.certificate.subject.to_s.split("CN=")[1].split("/")[0]
```

Comment:
This seems a bit hacky to me ... Maybe instead of trying to compare `owner_id`s (which should be the same for each org) we should download the matching certs here and compare serials?